src/Core/System/Snippet/Service/TranslationConfigLoader.php

```diff
@@ -16,16 +22,46 @@
 #[Package('discovery')]
 class TranslationConfigLoader
 {
-    private const TRANSLATION_CONFIG_DIR = __DIR__ . '/../../Resources';
-
-    private const TRANSLATION_CONFIG_FILE = '/translation.yaml';
+    public function __construct(
+        private readonly Filesystem $configReader,
+    ) {
+    }
 
-    public static function load(): TranslationConfig
+    public function load(): TranslationConfig
     {
-        $config = self::parseConfig();
+        $config = $this->parseConfig();
+
+        $urlString = $config['repository-url'];
+
+        if (!\is_string($urlString)) {
+            $exception = new \InvalidArgumentException('The repository-url in the translation config must be a string.');
+            try {
+                $encodedUrl = json_encode($urlString, \JSON_THROW_ON_ERROR);
+            } catch (\JsonException $e) {
+                $exception = $e;
+            }
+
+            throw SnippetException::invalidRepositoryUrl($encodedUrl, $exception);
```

Comment:
should probably be this
```suggestion
            throw SnippetException::invalidRepositoryUrl($urlString, $exception);
```